tests/lib/nested.sh

```diff
@@ -588,6 +588,25 @@ nested_configure_cloud_init_on_core20_vm() {
     kpartx -d "$IMAGE"
 }
 
+nested_save_serial_log() {
+    if [ -f "${NESTED_LOGS_DIR}/serial.log" ]; then
+        for i in $(seq 50); do
+            if [ ! -f "${NESTED_LOGS_DIR}/serial.log.${i}" ]; then
+                cp "${NESTED_LOGS_DIR}/serial.log" "${NESTED_LOGS_DIR}/serial.log.${i}"
+                break
+            fi
+        done
+    fi
+}
+
+nested_print_serial_log() {
+    if [ -f "${NESTED_LOGS_DIR}/serial.log" ]; then
+        for logfile in $(ls -r "${NESTED_LOGS_DIR}"/serial.log*); do
```

Comment:
The logs are quite long, given that there can be many, it will be very hard to follow. Given that we're stating a VM for each test, I'd rather have just the serial output of a test that run last (and failed if used in debug section).

Reply:
There's a fix proposed in https://github.com/snapcore/snapd/pull/9352/commits/fbb76cb4c81617449a64a0acaf26d3b41e968837

Reply:
The logs are deleted during the restore of each test, the idea of this is because during a test sometimes we start and stop the nested-vm service and when we start it the log from the previous start is overwritten. 
Currently we stop the vm to save image and we restart it, so in that case we are lossing the serial log for the first boot.